### PR TITLE
DOCS-2089: Add board API code samples

### PR DESCRIPTION
**Pull request by andf-viam**
Add code samples to Board API GO methods, for:
- `Board` API itself
- `Analog` API
- `GPIOPin` API
- `DigitalInterrupt` API

Code samples sourced from live docs.viam.com reference page: https://docs.viam.com/components/board/#api

### CODE REVIEW

**Comment by andf-viam** [comment]
@cheukt let me know if you'd like to see any changes -- just more code samples, all in `//` comments. Thanks so much!

**Comment by cheukt** [comment]
tagging @randhid instead

File: components/board/board.go [hunk 1]
```diff
@@ -46,6 +46,68 @@ func Named(name string) resource.Name {
...
+//
+//	// Get the name of every Analog pin configured on the board.
+//	names := myBoard.AnalogNames()
+//
+// DigitalInterruptNames example:
+//
+//	myBoard, err := board.FromRobot(robot, "my_board")
+//
+//	// Get the name of every DigitalInterrupt configured on the board.
+//	names := myBoard.DigitalInterruptNames()
```
Comment (randhid) [inline components/board/board.go:83]: we will remove these once Status is gone - so I'd just not document them.

File: components/board/board.go [hunk 2]
```diff
@@ -75,6 +137,28 @@ type Board interface {
...
 // An Analog represents an analog pin that resides on a board.
+//
+// Read example:
+//
+//	myBoard, err := board.FromRobot(robot, "my_board")
+//
+//	// Get the analog pin "my_example_analog".
+//	analog, err := myBoard.AnalogByName("my_example_analog")
+//
+//	// Get the value of the digital signal "my_example_analog" has most recently measured.
```
Comment (randhid) [inline components/board/board.go:148]: analog signal

File: components/board/digital_interrupts.go [hunk 3]
```diff
@@ -18,6 +18,19 @@ type Tick struct {
...
+//
+// Value example:
+//
+//	myBoard, err := board.FromRobot(robot, "my_board")
+//
+//	// Get the DigitalInterrupt "my_example_digital_interrupt".
+//	interrupt, err := myBoard.DigitalInterruptByName("my_example_digital_interrupt")
+//
+//	// If using type basic: Get the amount of times this DigitalInterrupt has been interrupted with a tick.
+//	count, err := interrupt.Value(context.Background(), nil)
```
Comment (randhid) [inline components/board/digital_interrupts.go:30]: Type basic is a pi-only implementation. This is just how many times a digital interrupt has ticked.

File: components/board/digital_interrupts.go [hunk 4]
```diff
@@ -18,6 +18,19 @@ type Tick struct {
...
+//
+//	// If using type servo: Get the rolling average of the pulse width across each time the DigitalInterrupt is interrupted with a tick.
+//	rolling_avg, err := interrupt.Value(context.Background(), nil)
```
Comment (randhid) [inline components/board/digital_interrupts.go:33]: Remove

File: components/board/board.go [hunk 5]
```diff
@@ -46,6 +46,68 @@ func Named(name string) resource.Name {
...
+//	if di8, err := myBoard.DigitalInterruptByName("8"); err == nil {
+//	  interrupts = append(interrupts, di8)
+//	}
+//
+//	if di11, err := myBoard.DigitalInterruptByName("11"); err == nil {
+//	  interrupts = append(interrupts, di11)
+//	}
+//
+//	// Stream ticks on ticksChan from the listed digital interrupts on pins 8 and 11.
+//	err = myBoard.StreamTicks(context.Background(), interrupts, ticksChan, nil)
```
Comment (randhid) [inline components/board/board.go:110]: show how a ticksChan can be made.

**Review by randhid — CHANGES_REQUESTED** [review]

File: components/board/board.go [hunk 6]
```diff
@@ -46,6 +46,54 @@ func Named(name string) resource.Name {
...
+//
+//	if di8, err := myBoard.DigitalInterruptByName("8"); err == nil {
+//	  interrupts = append(interrupts, di8)
+//	}
+//
+//	if di11, err := myBoard.DigitalInterruptByName("11"); err == nil {
+//	  interrupts = append(interrupts, di11)
+//	}
+//
+//	// Show how a ticksChan can be made from the listed digital interrupts on pins 8 and 11.
```
Comment (randhid) [inline components/board/board.go:95]:
```suggestion

```

File: components/board/gpio_pin.go [hunk 7]
```diff
@@ -3,6 +3,66 @@ package board
...
+//	high := pin.Get(context.Background(), nil)
+//
+// PWM example:
+//
+//	myBoard, err := board.FromRobot(robot, "my_board")
+//
+//	// Get the GPIOPin with pin number 15.
+//	pin, err := myBoard.GPIOPinByName("15")
+//
+//	// Get if it is true or false that the state of the pin is high.
```
Comment (randhid) [inline components/board/gpio_pin.go:34]: This returns the duty cycle

Comment (andf-viam) [reply components/board/gpio_pin.go:34]:
```suggestion
//	// Returns the duty cycle.
```

**Review by randhid — APPROVED** [review]

**Pull request merged**